packages/jupyter-ai/jupyter_ai/reply_processor.py

```diff
@@ -0,0 +1,29 @@
+import asyncio
+from typing import Dict
+from jupyter_ai.handlers import ChatHandler
+from ray.util.queue import Queue
+
+
+class ReplyProcessor():
+    """A single processor to distribute replies"""
+
+    def __init__(self, handlers: Dict[str, ChatHandler], queue: Queue, log):
+        self.handlers = handlers
+        self.queue = queue
+        self.log = log
+
+    def process(self, message):
+        self.log.debug('Processing message %s in ReplyProcessor', message)
+        for handler in self.handlers.values():
+            if not handler:
+                continue
+            
+            handler.broadcast_message(message)
+            break
+
+    async def start(self):
+        while True:
+            if not self.queue.empty():
+                self.process(self.queue.get())
```

Comment:
I believe that the ray queue support the asyncio Queue API so you can await on `self.queue.get()`.